### PR TITLE
fix(oras)!: remove ORAS artifact references support

### PR DESCRIPTION
**Pull request by rchincha**
ORAS artifacts/references predated OCI dist-spec 1.1.0 which now has the same functionality and likely to see wider adoption.

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/storage/s3/s3_test.go [hunk 1]
```diff
@@ -450,170 +449,6 @@ func TestStorageDriverStatFunction(t *testing.T) {
...
-		body := []byte("this is an artifact")
-		digest = godigest.FromBytes(body)
-		buf = bytes.NewBuffer(body)
-		buflen = buf.Len()
-
-		_, n, err := imgStore.FullBlobUpload(repo, buf, digest)
-		So(err, ShouldBeNil)
-		So(n, ShouldEqual, buflen)
-
-		Convey("Get OCI Referrers - application/vnd.oci.image.manifest.v1+json", func(c C) {
```
Comment (andaaron) [inline pkg/storage/s3/s3_test.go:528]: Please double check if this code needs to be deleted.

File: swagger/docs.go [hunk 2]
```diff
@@ -1503,8 +1449,6 @@ var SwaggerInfo = &swag.Spec{
 	Description:      "APIs for Open Container Initiative Distribution Specification",
 	InfoInstanceName: "swagger",
 	SwaggerTemplate:  docTemplate,
-	LeftDelim:        "{{",
```
Comment (andaaron) [inline swagger/docs.go:1506]: Double check your swag version you used matches the one in the makefile, I suspect it doesn't.

**Review by andaaron — APPROVED** [review]

**Review by peusebiu — APPROVED** [review]
lgtm

File: test/blackbox/sync_docker.bats [hunk 3]
```diff
@@ -332,7 +332,7 @@ function teardown_file() {
 
 @test "sync image on demand from ghcr.io" {
     zot_port=`cat ${BATS_FILE_TMPDIR}/zot.port`
-    run skopeo copy docker://127.0.0.1:${zot_port}/project-zot/zot-linux-amd64:v2.0.0-rc5 oci:${TEST_DATA_DIR} --src-tls-verify=false
```
Comment (andaaron) [inline test/blackbox/sync_docker.bats:335]: You need to also update the assertions the tests make

File: test/blackbox/sync_docker.bats [hunk 4]
```diff
@@ -450,7 +450,7 @@ function teardown_file() {
 
 @test "run docker with image synced from ghcr.io" {
     zot_port=`cat ${BATS_FILE_TMPDIR}/zot.port`
-    run docker run -d 127.0.0.1:${zot_port}/project-zot/zot-linux-amd64:v2.0.0-rc5
+    run docker run -d 127.0.0.1:${zot_port}/project-zot/zot-linux-amd64:v2.0.1
```
Comment (andaaron) [inline test/blackbox/sync_docker.bats:453]: Update also line 461

File: test/blackbox/sync_docker.bats (hunk 3 above)
Comment (andaaron) [reply test/blackbox/sync_docker.bats:335]: Line 343

**Review by andaaron — APPROVED** [review]

**Pull request merged**